### PR TITLE
Add popup on poi hover

### PR DESCRIPTION
**Pull request by Julien-laville**
selection_005

### CODE REVIEW

File: src/adapters/osm_schedule.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  return result;
+}
+
+
+function scheduleStatus(schedule, timeMessages) {
+  if(!schedule) {
+    return {msg : '', color : '#fff'}
+  }
+  let remaining = schedule.seconds_before_next_transition
```
Comment (amatissart) [inline src/adapters/osm_schedule.js]: I think there is a confusion here between `scheduleResponse` and `schedule` that has no "seconds_before_next_transition" defined. And as a result, the status is now always "open" :stuck_out_tongue: 

Also, maybe we could completely get rid of the client-side `parse` function, now that the API returns complete "days" schedules (with potentially multiple opening periods per day).

File: src/adapters/poi_popup.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  this.map.on('mouseleave', layer, async () => {
+    this.isCounting = false
+    clearTimeout(this.timeOutHandler)
+  })
+}
+
+PoiPopup.prototype.create = async function (layerPoi) {
+  let poi = await Poi.poiApiLoad(layerPoi.properties.global_id)
+  if(poi) {
+    let {color} = IconManager.get(poi)
```
Comment (amatissart) [inline src/adapters/poi_popup.js]: This breaks when the `poi` has no specific icon, (and `IconManager.get` returns undefined).  
We need a default color in that case.

File: src/views/popup.dot [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  {{? this.opening && this.opening.schedule }}
+    <span class="poi_popup__opening">
+    {{= _(this.opening.status.msg) }} {{? this.opening.nextTransition }} - {{= _('until {nextTransitionTime}', 'hour panel', {nextTransitionTime : this.opening.nextTransition})  }}{{?}}
+    {{? this.opening.isTwentyForSeven }}
+      24 / 7
+    {{?}}
+    </span>
+    <div class="poi_popup__title__hour__circle" style="background:{{= this.opening.status.color }}"></div>
+  {{?? this.address}}
+    <span class="poi_popup__address"{{= this.address }}</span>
```
Comment (amatissart) [inline src/views/popup.dot]: The opening  "span" is not closed, `>` is missing.

File: src/adapters/poi_popup.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+
+function PoiPopup() {}
+
+PoiPopup.prototype.init = function(map) {
+  this.map = map
+  this.popupHandle = null
+
+  this.timeOutHandler = null
+  this.isCounting = false
```
Comment (amatissart) [inline src/adapters/poi_popup.js]: This boolean `isCounting` can be removed, `timeOutHandler` is enough to implement the listener behaviour.

**Comment by amatissart** [comment]
Some general comments after my review:
* the popup needs more specification. For example: 
  * Should multiple popups be visible at the same time (as it is in the current implementation) ? 
  * Should a popup be triggered when hovering the POI that is currently selected in the panel ?
* it is probably needed to implement an API endpoint that would return only relevant blocks for the popup ? Or ensure the response is kept in cache, so that a second request is not triggered on click ?

**Comment by nlehuby** [comment]
There is an issue with the opening hours `closed` status 
screenshot_2018-08-14 qwant maps - beta
:thinking: 

And the last day of the week is not called `Undefined` :yum:

**Comment by nlehuby** [comment]
and the opening_hours block is cut when there is too much intervals to display:
screenshot_2018-08-14 qwant maps - beta 1

**Comment by Julien-laville** [comment]
simple
fix css styling & opeing parsing

File: src/adapters/poi_popup.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      return poiConfig.apiName === 'opening_hours'
+    })
+    let opening
+    let address
+    if(hours) {
+      opening = new OsmSchedule(hours, timeMessages.options.messages)
+    } else if(poi.address){
+      address = poi.address.label
+    }
+    this.popupHandle = new Popup({className: 'poi_popup__container', closeButton : false, closeOnClick : true, offset : {'bottom-left' : [18, -8]}, anchor : 'bottom-left'})
```
Comment (amatissart) [inline src/adapters/poi_popup.js]: The popup anchor shoud take the position on the screen into account.  
It should be entirely visible, even when hovering a POI near the edges of the screen:
capture du 2018-08-16 12-11-05

File: src/panel/poi_bloc/hour_panel.js [hunk 6]
```diff
@@ -1,98 +1,18 @@
...
-import openingHourParse from '../../../src/adapters/opening_hour'
-import I18n from '../../libs/i18n'
-const strftime = require('strftime')
+import OsmSchedule from '../../../src/adapters/osm_schedule'
 
 function HourPanel(block, poi, options) {
   this.panel = new Panel(this, HourPanelView)
   this.name = block.name
-  this.timeMessages = options.messages
+  this.timeMessages =
```
[the remark this answers is not in the document]
Comment (Julien-laville) [reply src/panel/poi_bloc/hour_panel.js]: i missed my line deletion

File: src/scss/includes/mapbox-gl-override.scss [hunk 7]
```diff
@@ -13,30 +13,10 @@
     display: none;
 }
 
-.mapboxgl-canvas-container.mapboxgl-interactive,
```
[the remark this answers is not in the document]
Comment (Julien-laville) [reply src/scss/includes/mapbox-gl-override.scss]: yep, i'll only keep mapboxgl-canary & mapboxgl-ctrl-bottom-*

File: src/scss/includes/popup.scss [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+
+.poi_popup {
+  min-width: 300px;
+  border-radius: 4px;
+  background: #fff;
+  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.1);
+  border-left: 4px solid;
+  padding: 20px 0;
```
[the remark this answers is not in the document]
Comment (Julien-laville) [reply src/scss/includes/popup.scss]: sure

File: src/views/poi_bloc/hour.dot [hunk 9]
```diff
@@ -1,24 +1,33 @@
...
-          {{? !this.hours['24/7'] }}
+          {{? !this.opening.isTwentyForSeven }}
             <table class="poi_panel__info__hours__table">
-              {{~ this.displayHours:day:i }}
+              {{~ this.opening.displayHours:day:i }}
                 <tr {{= i + 1 === new Date().getDay() ? 'class="poi_panel__info__hours--current"' : '' }}>
                   <td class="day">{{= day.dayName }}</td>
-                  <td class="hours">{{= day.opening ?  (day.opening + '').replace(',', ' - ') : _('Closed')}}<td>
+                  <td class="hours">
+                    {{ console.log(day) ; }}
```
Comment (nlehuby) [inline src/views/poi_bloc/hour.dot]: :scream_cat:

File: src/adapters/poi_popup.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+PoiPopup.prototype.create = async function (layerPoi, event) {
+  let poi = await Poi.poiApiLoad(layerPoi.properties.global_id)
+  if(poi) {
+    if(this.popupHandle) {
+      this.popupHandle.remove()
+      this.popupHandle = null
+    }
+    let {color} = IconManager.get(poi)
```
Comment (nlehuby) [inline src/adapters/poi_popup.js]: if the POI has the default picto and color, it errors and no popup is shown

**Comment by nlehuby** [comment]
rebased and merged

**Pull request closed without merging**